hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/jdbc/SelectProcessorSupplier.java

```diff
@@ -31,27 +32,73 @@
 import javax.annotation.Nullable;
 import java.io.IOException;
 import java.security.Permission;
-import java.sql.Date;
 import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
 import java.sql.SQLException;
-import java.sql.Time;
-import java.sql.Timestamp;
-import java.util.ArrayList;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.OffsetDateTime;
 import java.util.Collection;
+import java.util.HashMap;
 import java.util.List;
+import java.util.Locale;
+import java.util.Map;
 
 import static com.hazelcast.security.permission.ActionConstants.ACTION_READ;
+import static java.util.Collections.singleton;
 import static java.util.Collections.singletonList;
 import static java.util.Objects.requireNonNull;
 
+@SuppressWarnings("checkstyle:ExecutableStatementCount")
 public class SelectProcessorSupplier
         extends AbstractJdbcSqlConnectorProcessorSupplier
         implements ProcessorSupplier, DataSerializable, SecuredFunction {
 
+    private static final Map<String, BiFunctionEx<ResultSet, Integer, Object>> GETTERS = new HashMap<>();
```

Comment:
Instead of copying the value getters for each processor
1. Create a case insensitive map e.g
`Map<String, Integer> treeMap = new TreeMap<>(String.CASE_INSENSITIVE_ORDER);`
or use the from Apache Commons CaseInsensitiveMap
2. In the result set processing lamda do this. I did not test it but this should give you the getter for the column type
````
(rs) -> {
                    ResultSetMetaData metaData = rs.getMetaData();
                    int columnCount = rs.getMetaData().getColumnCount();
                    Object[] row = new Object[columnCount];
                    for (int index = 0; index < columnCount; index++) {
                        String type = metaData.getColumnTypeName(index + 1).toUpperCase(Locale.ROOT);
                        BiFunctionEx<ResultSet, Integer, Object> fn = GETTERS.getOrDefault(
                                type,
                                (resultSet, n) -> rs.getObject(n)
                        );
                        row[index] = fn.apply(rs,index);
                    }
                   
                    return new JetSqlRow(evalContext.getSerializationService(), row);
                }
````

Reply:
This would definitely simplify the solution and would be correct, but it would execute
```
                        String type = metaData.getColumnTypeName(index + 1).toUpperCase(Locale.ROOT);
                        BiFunctionEx<ResultSet, Integer, Object> fn = GETTERS.getOrDefault(
                                type,
                                (resultSet, n) -> rs.getObject(n)
                        );
```
for each column*row.